frontend/public/components/factory/table.tsx

```diff
@@ -276,15 +276,52 @@ export type TableRowProps = {
   className?: string;
 };
 
+const BREAKPOINT_SM = 576;
+const BREAKPOINT_MD = 768;
+const BREAKPOINT_LG = 992;
+const BREAKPOINT_XL = 1200;
+const BREAKPOINT_XXL = 1400;
+const MAX_COL_SM = 2;
+const MAX_COL_MD = 4;
+const MAX_COL_LG = 4;
+const MAX_COL_XL = 6;
+const MAX_COL_XXL = 8;
+
+const isColumnVisible = (columnID: string, columns: Set<string> = new Set()) => {
+  const showNamespace =
+    columnID !== 'namespace' || UIActions.getActiveNamespace() === ALL_NAMESPACES_KEY;
+  if (_.isEmpty(columns) && showNamespace) {
+    return true;
+  }
+  if (!columns.has(columnID)) {
+    return false;
+  }
+  const widthInPixels = window.innerWidth;
+  if (widthInPixels < BREAKPOINT_SM) {
+    return [...columns].indexOf(columnID) < MAX_COL_SM;
```

Comment:
I'm confused by this since `columns` is a `Set`. Does `Set` have a guaranteed ordering?

nit: Calculate the index once and reuse it for each check.

```ts
const columnIndex = [...columns].indexOf(columnID);
```

Reply:
@spadgett you are right about the set ordering...looking into that now. Fixed nit.